meshapp/src/components/SideMenu.tsx

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import {
+  Drawer,
+  Grid,
+  IconButton,
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemIcon,
+  ListItemText,
+  Toolbar,
+} from "@mui/material";
+import {
+  TableRows as MenuIcon,
+  Notifications as NotificationsIcon,
+  Home as HomeIcon,
+  Receipt as ProfileIcon,
+  Inbox as MessagingIcon,
+  People as SwipingIcon,
+  Settings as SettingsIcon,
+  Logout as LogoutIcon,
+} from "@mui/icons-material";
+
+type ListItemTypes = { text: String; Icon: React.ElementType };
+
+// TODO: Add routes to each menu item when routing is implemented
+const ListItemComponent = ({ text, Icon }: ListItemTypes) => (
+  <ListItem>
+    <ListItemButton
+      sx={{ borderBottom: "1px solid", borderColor: "text.secondary" }}
```

Comment:
Though the figma does say text/secondary, it doesn't look like the dark theme part matches the figma, and the text is a bit hard to see. Do you think we should add another theme color to the palette to use specifically for the menu bar?

Reply:
I left notes about the theme changes in my pr description. Additionally I think that palette changes aren't in the scope of the task and should be addressed in another issue.